docs/repositories-configure/using-submodules.md

```diff
@@ -2,39 +2,53 @@
 
-    When you add a repository to Codacy, Codacy adds a new SSH key to your repository using an integration with your Git provider. Codacy uses this SSH key every time it needs to clone that repository. When you're using submodules, Codacy needs to clone additional repositories it has no access to. Since there is no way of identifying the submodules of a repository before cloning the repository, Codacy adds the SSH keys to the user account so that it has access to the same repositories as the user.
+1.  Add a new SSH key to your git provider account by clicking the link **Add new user key** or the button **Generate New User Key**, depending on your Git provider.
+
+    For GitHub and Bitbucket, this takes you to the Git provider page where you can manage your user account SSH keys. For GitLab, Codacy removes the existing repository key and creates the new SSH key on your user account automatically.
 
     ![Generate new user key](images/using-submodules-generate-new-user-key.png)
 
 1.  If you're using submodules to share an analysis tool configuration file across your repositories, check if your tool recursively searches the subdirectories of your repositories for configuration files.
 
     If your tool doesn't detect the configuration files in the submodule directories, you must include a configuration file directly in the root of your repositories referencing the configuration files in the submodule directories.
 
-## Using submodules on new repositories
+## Automating user keys for new repositories
 
-You must repeat steps 2 to 4 above every time you add new repositories with submodules to Codacy.
+!!! info "This section applies only to Codacy Self-hosted"
 
-Alternatively, you can make this the default behavior for all new repositories by enabling **Add project key to the user, by default** in **Administration**, page **Settings**.
+You can set Codacy to automatically add the new SSH key to your git provider account for all new repositories by enabling **Add project key to the user, by default** in **Administration**, page **Settings**.
 
 !!! important
     **If you're using Bitbucket Cloud** this setting must be turned off since automatically adding the user keys isn't supported.
 
 ![Add project key to the user by default](images/using-submodules-default-add-user-key.png)
+
+## See also
+
+-   [Managing deploy keys in GitHub](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/managing-deploy-keys)
+-   [Add an SSH key to your GitHub account](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent)
+-   [Configure repository settings in Bitbucket](https://support.atlassian.com/bitbucket-cloud/docs/configure-repository-settings/)
+-   [Add an SSH key to your Bitbucket account](https://support.atlassian.com/bitbucket-cloud/docs/configure-ssh-and-two-step-verification/)
```

Comment:
It's a great idea to provide these references here, so they don't overcomplicate the instructions above! :ok_hand: